scripts/check_frontend_test_coverage.py

```diff
@@ -0,0 +1,430 @@
+        sys.exit(1)
+
+
+def filter_relevant_lines(line):
+    """Check if the line has file path (SF) or total lines (LF) or covered
+    lines (LH) of the test.
+
+    Args:
+        line: str. A line from lcov file.
+
+    Returns:
+        Boolean. If the line has the file path or total lines or covered lines
+        of the test.
+    """
+    return (
+        line.startswith('SF') or line.startswith('LH')
+        or line.startswith('LF'))
+
+
+def get_stanzas_from_lcov_file():
+    """Get all stanzas from a lcov file. The lcov file gather all the frontend
+    files that has tests and each one has the following structure:
+    TN: test name
+    SF: file path
+    FNF: total functions
+    FNH: functions covered
+    LF: total lines
+    LH: lines covered
+    BRF: total branches
+    BRH: branches covered
+    end_of_record
+
+    Returns:
+        list(Stanza). A list with all stanzas.
+    """
+
+    f = python_utils.open_file(LCOV_FILE_PATH, 'r')
+    lcov_items_list = f.read().split('end_of_record')
+    stanzas_list = []
+
+    for item in lcov_items_list:
+        relevant_lines = [line for line in item.splitlines() if 
+            filter_relevant_lines(line)]
+
+        if len(relevant_lines) > 0:
+            stanza = LcovStanzaRelevantLines()
+
+            stanza.set_file_name(relevant_lines[0])
+            stanza.set_total_lines(relevant_lines[1])
+            stanza.set_covered_lines(relevant_lines[2])
+
+            stanzas_list.append(stanza)
+
+    return stanzas_list      
+
+
+def remove_item_from_whitelist(whitelist, test_name):
+    """Remove an item from the whitelist.
+    
+    Args:
+        whitelist: list(str). The whitelist including only fully covered tests.
+        test_name: str. The test to be removed from whitelist.
+    """
+    index = whitelist.index(test_name)
+    whitelist.pop(index)
+
+
+def check_fully_covered_filenames_sorting():
+    if fully_covered_filenames != sorted(fully_covered_filenames):
+        python_utils.PRINT(
+            'The \033[1mfully_covered_filenames\033[0m list must be kept'
+            ' in alphabetic order.')
+        sys.exit(1)
+
+
+def check_coverage_changes():
+    """Checks if the whitelist for fully covered files needs to be changed by:
+    - New file insertion
+    - File renaming
+    - File deletion
+
+    Raises:
+        Exception: If LCOV_FILE_PATH doesn't exist.
+    """
+    if not os.path.exists(LCOV_FILE_PATH):
+        raise Exception('File at path {} doesn\'t exist'.format(
+            LCOV_FILE_PATH))
+
+    stanzas = get_stanzas_from_lcov_file()
+    whitelist = list(fully_covered_filenames)
+    errors = ''
+
+    for stanza in stanzas:
+        file_name = stanza.get_file_name()
+        total_lines = stanza.get_total_lines()
+        covered_lines = stanza.get_covered_lines()
+
+        if file_name in whitelist:
+            if total_lines != covered_lines:
+                errors += ('\033[1m{}\033[0m file is in the whitelist but its'
```

Comment:
in the whitelist --> supposed to have 100% coverage

Reply:
Done